src/nccl_ofi_rdma.c

```diff
@@ -5379,6 +5410,10 @@ static int release_ep(nccl_net_ofi_ep_t *base_ep)
 	 * deallocation.
 	 */
 	if (ep->ref_cnt == 0) {
+
+		if (ofi_nccl_endpoint_per_communicator() != 0)
+			nccl_ofi_delete_ep_for_addr(device->ep_addr_list, &ep->base);
```

Comment:
For non-thread-local-eps, also free the endpoint, since the comment above about the thread-local pointer preventing freeing doesn't apply.